sklearn/model_selection/_split.py

```diff
@@ -51,12 +53,30 @@
 ]
 
 
-class BaseCrossValidator(metaclass=ABCMeta):
+class GroupsComsumerMixin(_MetadataRequester):
+    """A Mixin to ``groups`` by default.
+
+    This Mixin makes the object to request ``groups`` by default as
+    ``REQUESTED``, instead of the ``ERROR_IF_PASSED`` default value.
+
+    .. versionadded:: 1.1
+    """
+
+    __metadata_request__split = {"groups": RequestType.REQUESTED}
+
+
+class BaseCrossValidator(_MetadataRequester, metaclass=ABCMeta):
     """Base class for all cross-validators
 
     Implementations must define `_iter_test_masks` or `_iter_test_indices`.
     """
 
+    # This indicates that by default CV splitters don't have a "groups" kwarg,
+    # unless indicated by inheriting from ``GroupsComsumerMixin``.
+    # This also prevents ``set_split_request`` to be generated for splitters
+    # which don't support ``groups``.
+    __metadata_request__split = {"groups": RequestType.UNUSED}
```

Comment:
My mind is a little blown. So the current API for `KFold.split` has a `groups` parameter it never uses. Because of the inheritance structure, the [API docs for KFold.split](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.KFold.html#sklearn.model_selection.KFold.split) says the group labels are used in `KFold`.

(This comment is not actionable)